lib/checkpoint-writer.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+const EventEmitter = require('events');
+const _ = require('lodash');
+let writerLoopStopped = true;
+let lastDoc;
+let lastCheckpointId;
+let harvesterApp;
+let writeInterval = 1;
+const log = require('agco-logger')({});
+const shouldLog =  (process.env.LOG_EVENTS_READER=="true") ||false;
+
+const checkpointEvent = new EventEmitter();
+
+checkpointEvent.on('newCheckpoint', (checkpointId, doc) => {
+  lastCheckpointId = checkpointId;
+  lastDoc = doc;
+});
+
+const persistLastCheckpoint = () => {
+  if(lastDoc && lastCheckpointId) {
+    const profileMessage = 'writing new checkpoint to database ' + lastDoc.ts;
+
+    shouldLog && log.profile(profileMessage);
+    harvesterApp.adapter.update('checkpoint', lastCheckpointId, {ts: lastDoc.ts})
+      .then(() => {
+        shouldLog && log.profile(profileMessage);
+      })
+      .catch(error => {
+        console.log(error);
+        process.exit(1);
+      });
+  };
+  lastCheckpointId = null;
+  lastDoc = null;
+};
+
+const persistInInterval = () => {
+  setTimeout(() => {
+    persistLastCheckpoint();
+    persistInInterval();
+  }, writeInterval);
+};
+
+const startWriterLoop = (app, loopStopped) => {
+  loopStopped = loopStopped || writerLoopStopped;
```

Comment:
We need this boolean to know if the loop is running or not, and we need to have an accessor to change its state because Mocha stops the setInterval after each test case. But we will put an accessor instead of receive it as parameter.